### PR TITLE
MGMT-9779: CI - Retry cluster-info readiness

### PR DESCRIPTION
**Pull request by nmagnezi**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: tools/wait_for_cluster_info.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+SLEEP = 10
+
+
+def handle_arguments():
+    return deployment_options.load_deployment_options()
+
+
+@retry(exceptions=RuntimeError, tries=2, delay=3)
+def is_cluster_info_ready(namespace, kubectl_cmd) -> bool:
+    print(f"DEBUG - is cluster-info ready (kubectl_cmd: {kubectl_cmd} namespace: {namespace} )")
```
Comment (omertuc) [inline tools/wait_for_cluster_info.py:18]: `namespace` is unused

File: Makefile [hunk 2]
```diff
@@ -257,7 +257,7 @@ $(KUBECTL) rollout status  deployment assisted-service
 endef
 
 _verify_cluster:
-	$(KUBECTL) cluster-info
```
Comment (omertuc) [inline Makefile:260]: Why is the `namespace` needed here?

File: tools/wait_for_cluster_info.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+def main():
+    deploy_options = handle_arguments()
+    if not deploy_options.apply_manifest:
+        return
+
+    waiting.wait(
+        lambda: is_cluster_info_ready(
+            namespace=deploy_options.namespace,
+            kubectl_cmd=utils.get_kubectl_command(deploy_options.target, deploy_options.namespace)
```
Comment (omertuc) [inline tools/wait_for_cluster_info.py:37]: (`get_kubectl_command` is OK with `namespace=None`, and it's the default value, so it's not really needed here either)

Comment (nmagnezi) [reply tools/wait_for_cluster_info.py:37]: what about custom namespaces? as we use in e2e jobs?

Comment (omertuc) [reply tools/wait_for_cluster_info.py:37]: cluster-info doesn't need any namespace

Comment (nmagnezi) [reply tools/wait_for_cluster_info.py:37]: This is how we used it, so I assumed as such. But looks like indeed this is the case. I'll remove it.

File: tools/wait_for_cluster_info.py (hunk 1 above)
Comment (nmagnezi) [reply tools/wait_for_cluster_info.py]: Removed

File: Makefile (hunk 2 above)
Comment (nmagnezi) [reply Makefile:260]: Removed

File: tools/wait_for_cluster_info.py (hunk 3 above)
Comment (nmagnezi) [reply tools/wait_for_cluster_info.py]: Done

**Review by osherdp — APPROVED** [review]

**Pull request merged**